### ISSUE TITLE
Boats spawn underwater

### ISSUE THREAD

**Issue by Aivar-RoR** [RigsOfRods/rigs-of-rods#147]
Boats that are spawned via the port building are spawned underwater, the same thing happens with boats that are placed on terrain. Boats that are underwater get deformed, some can be fixed with backspace, some cannot.
Tried with North St. Helens, Island, Starling Island.

**Comment by only-a-ptr** [comment]
@Aivar-RoR : Please post me download links to the boats you tested with, along with comments how exactly they behave (the backspace key etc...)

**Comment by Aivar-RoR** [comment]
C1 boat: http://www.rigsofrods.com/repository/view/1227
Weasel Boat 0.4.0.7 version: http://www.rigsofrods.com/repository/view/5301
Houseboat: http://www.rigsofrods.com/repository/view/3764
Wahoo: http://www.rigsofrods.com/repository/view/3746
Atlas tug: http://www.rigsofrods.com/repository/view/5300
All of these boats spawn underwater when spawned via walking into the port building or when the are placed on a terrain. When underwater, the boats deform. Hitting backspace a few times rotates them, raises them to the surface and fixes the deformation. 
Smit Rotterdam tug: http://www.rigsofrods.com/repository/view/3749
Spawns underwater, usually with 100% destruction. Cannot be fixed with the backspace key: this just rotates the tug, but it doesn't surface nor fix the tug. Happens both when spawned via the port building or when placed on a terrain.

All of the boats work fine when spawned from the in-game menu.

Tried also in test build 3.

**Comment by only-a-ptr** [comment]
@Aivar-RoR I'm assuming the "starling island" you refer to is this conversion: http://www.rigsofrods.com/threads/109425-Updated-terrains-for-0-4-0-7?p=1260243&viewfull=1#post1260243 >> RoR 0.4.5 Terrains

**Comment by only-a-ptr** [comment]
@Aivar-RoR : Okay I started with "Starling island". I downloaded all the content you linked above. I observe 2 sunken boats:
1. The weasel. Spawns completely on the bottom, no damage. When entered, RoR goes haywire: game lags, camera is placed all wrong, backspace doesn't work..
![screenshot_8](https://cloud.githubusercontent.com/assets/491088/11312352/d409d046-8fd5-11e5-9a03-bbf01aec8ea3.jpg)


2. The Smit. Spawns without damage, lies on the bottom and there are water splashes falling from the sky as if it just sank. Can be entered normally, except it won't sail. backspace works, but the boat always sinks back down. Controls also work, but the boat only moves until it hits bottom.
![screenshot_9](https://cloud.githubusercontent.com/assets/491088/11312331/c17ac822-8fd5-11e5-8d29-12f7f3706a83.jpg)

Anything I missed?

**Comment by DarthCainRor** [comment]
Not certain here, but I believe he may be talking about the boat spawner building.
![screenshot_194](https://cloud.githubusercontent.com/assets/10333582/11317168/fca50382-8fe8-11e5-9590-34147e6e8e0a.jpg)
The boat spawn menu appears when entering this door.
![screenshot_195](https://cloud.githubusercontent.com/assets/10333582/11317169/0214fbc4-8fe9-11e5-8678-251e88d9dad7.jpg)

**Comment by Aivar-RoR** [comment]
@DarthCainRor  Indeed, I'm talking about the building you mentioned.
@only-a-ptr I observe the same behaviour you described (both the 0.4.0.7 and 0.4.5.1 descriptions). RoR hasn't always behaved like this. 
A few screenshots from 0.39.5 (the only version I have installed besides 0.4.0.7 and 0.4.5.1) and 0.4.5.1.
![smit_0 39](https://cloud.githubusercontent.com/assets/11720273/11318530/5dcb1ba8-905f-11e5-9b47-57efeebfa79b.jpg)
![smit_0 4](https://cloud.githubusercontent.com/assets/11720273/11318534/6a0c8dc0-905f-11e5-8fcc-3272038a99d8.jpg)
![weasel_0 39](https://cloud.githubusercontent.com/assets/11720273/11318535/6ef83fd2-905f-11e5-8d4c-cf9786981262.jpg)
![weasel_0 4](https://cloud.githubusercontent.com/assets/11720273/11318536/7444ce9c-905f-11e5-82fa-f45a12e41df8.jpg)
As you can see the aforementioned boats worked fine in 0.39.5. 
The Smit tug spawned fine from RoR versions 0.32 to 0.39.7. (I have tested every single RoR version since 0.32 throughout the years).  
It seems as if something has changed in the way how RoR handles boat spawning. Perhaps this behaviour has to do with the underwater camera that was added in 0.4.x (cannot remember exactly)?

Similarly, a few years ago some change in RoR's code broke truck spawning (or simply changed something) via the .terrn file:  trucks spawned in the air. Then the truck2 option was added so one could spawn the truck on the ground.

**Comment by only-a-ptr** [comment]
@Aivar-RoR Thanks for those clues into RoR's history, they help me a lot. From now on, you're my "go-to" person regarding RoR's past versions.

The cardinal issue here is that v0.4+ and v0.38 (v0.39?) ran in parallel for a while. I never tracked the exact moment. One example of this divergence: Aperion's "engine spasm fix" patch: http://www.rigsofrods.com/threads/54676-engine-spasm-fix?p=571718&viewfull=1#post571718. Added to codebase here: https://github.com/only-a-ptr/ror-legacy-svn-trunk/commit/cd163836a828ae27dd4f4a7bb96dbbc746f1eed1. Not present in v0.4 branch.

**Comment by Hiradur** [comment]
> "RoR Development History Timeline"

It would also be nice to have an overview of features which old versions have while newer versions don't.

**Comment by only-a-ptr** [comment]
@Hiradur I count that into the "timeline" concept ... chronologically list all features, their inception and eventual removal. With notes on historical context and eventual unsolicited changes in behavior.

Such a listing might get fuzzy. For example, there's Aperion's "engine spasm fix" (truckfile keyword ```patchEngineTorque```) http://www.rigsofrods.com/threads/54676-engine-spasm-fix?p=571718&viewfull=1#post571718 added in 2009 and used, for example, by famous Tatra-Dakar. For some reason, it was removed by Ulteq in https://github.com/RigsOfRods/rigs-of-rods/commit/480afed2b206b9514ad4c18bcf363f8a19f8412d. The question is - did Ulteq remove a working feature, or was the feature already obsoleted by something else? There were extensions to ```engoption``` in the meantime.

**Comment by Aivar-RoR** [comment]
@only-a-ptr Will start a thread about the timeline once I'm finished with the translation in Transifex.

**Comment by Aivar-RoR** [comment]
Started the thread at the forums.
http://www.rigsofrods.com/threads/120667-RoR-Development-History-Timeline

**Comment by only-a-ptr** [comment]
Browsing code
> 0.39.5
>> Preloading vehicles: https://github.com/only-a-ptr/ror-legacy-svn-trunk/blob/c01d3eacb66406d9defd05700fde3cf8d4af1ffe/source/main/gameplay/RoRFrameListener.cpp#L5071
>> Spawning vehicles: https://github.com/only-a-ptr/ror-legacy-svn-trunk/blob/c01d3eacb66406d9defd05700fde3cf8d4af1ffe/source/main/gameplay/RoRFrameListener.cpp#L5331

> 0.4.5
>> Preloading: https://github.com/RigsOfRods/rigs-of-rods/blob/d3e937442b7a276cd75709eff9b4ac83033458aa/source/main/terrain/TerrainObjectManager.cpp#L517
>> Spawning: https://github.com/RigsOfRods/rigs-of-rods/blob/d3e937442b7a276cd75709eff9b4ac83033458aa/source/main/terrain/TerrainObjectManager.cpp#L1196

It's mostly the same code, just moved around

**Comment by Hiradur** [comment]
today's observation: If you set the y coordinate of pre-spawned boats to 0 in the corresponding *.tobj file of the map then they spawn at water level, tested with different maps and boats.
So obviously a preset value for y (height) is screwing up the positioning. Gotta find out where the problem is but that's for another day.

I can't test with the boat spawner building due to #462
